js/wp-autoupdates.js

```diff
@@ -1,286 +1,88 @@
-				_ajax_nonce: href._wpnonce,
-				type: 'theme',
-				asset: href.theme
-			},
-			function (response) {
-
-			}
-		)
-		.done(function (response) {
-			if ( response.success ) {
-				$parent.html( response.data.return_html );
-				$parent.find('.theme-autoupdate-enable').focus();
-				wp.a11y.speak( wp_autoupdates.auto_disabled, 'polite' );
-			} else {
-				var errorHTML = add_error_notice( html, response.data.error );
-				wp.a11y.speak( response.data.error, 'polite' );
-				$parent.html( errorHTML );
-			}
-		})
-		.fail(function (response) {
-			var errorHTML = add_error_notice( html, wp_autoupdates.auto_update_error );
-			wp.a11y.speak( wp_autoupdates.auto_update_error, 'polite' );
-			$parent.html( errorHTML );
-		})
-		.always(function (response) {
-		});
-	});
-	// Enable auto-updates for a theme.
-	$('.theme-overlay').on('click', 'a.theme-autoupdate-enable', function (e) {
-		e.preventDefault();
-		var $anchor = $( this );
-		$anchor.blur();
-		var href = wpAjax.unserialize( $anchor.attr( 'href' ) );
-		var $parent = $anchor.parents( '.theme-autoupdate' );
-		// Clear errors
-		$parent.find( '.notice' ).remove();
-		var html = $parent.html();
-
-		// Show loading status.
-		$parent.find( '.theme-autoupdate-label' ).html( '<span class="dashicons dashicons-update spin"></span> ' + wp_autoupdates.enabling );
-
-		$.post(
-			ajaxurl,
-			{
-				action: 'enable_auto_updates',
-				_ajax_nonce: href._wpnonce,
-				type: 'theme',
-				asset: href.theme
-			},
-			function (response) {
-
-			}
-		)
-		.done(function (response) {
-			if ( response.success ) {
-				$parent.html( response.data.return_html );
-				$parent.find('.theme-autoupdate-disable').focus();
-				wp.a11y.speak( wp_autoupdates.auto_enabled, 'polite' );
-			} else {
-				var errorHTML = add_error_notice( html, response.data.error );
-				wp.a11y.speak( response.data.error, 'polite' );
-				$parent.html( errorHTML );
-			}
-			
-		})
-		.fail(function (response) {
-			var errorHTML = add_error_notice( html, wp_autoupdates.auto_update_error );
-			wp.a11y.speak( wp_autoupdates.auto_update_error, 'polite' );
-			$parent.html( errorHTML );
-		})
-		.always(function (response) {
-		});
-	});
-});
\ No newline at end of file
+		label.text( 'enable' === action ? wp_autoupdates.enabling : wp_autoupdates.disabling );
+		anchor.find( '.dashicons-update' ).removeClass( 'hidden' );
+
+		data = {
+			action: 'toggle_auto_updates',
+			_ajax_nonce: href._wpnonce,
+			state:  action,
+			type: type,
+			asset: 'plugin' === type ? href.plugin : href.theme,
+		};
+
+		$.post( ajaxurl, data )
+			.done( function( response ) {
+				if ( response.success ) {
+					// Update the counts in the enabled/disabled views if on on
+					// screen with a list table.
+					// TODO: If either count started out 0 the appropriate span won't
+					//       be there and hence won't be updated.
+					if ( 'themes' !== pagenow ) {
+						var enabled        = $( '.autoupdate_enabled span' ),
```

Comment:
JS coding standards: `var` statement should be at the beginning of the function https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/#declaring-variables-with-var

Reply:
I'm planning on raising that during `Open Floor` of `devchat` tomorrow.  Yes, by the "letter of the law" having a `var` at the beginning of a block is "wrong"; but by the "spirit of the law" (i.e., prevent polluting the global scope) it is fine.  So, I think the JSCS should be modified.

Thoughts?

Reply:
I think it's not correct because JavaScript has something called "hoisting" 🙂 The `var` declarations will be "moved" at the top of their scope anyways during the compile phase. `var` is function-scoped. This is why the coding standards ask to have them at the top: to avoid unexpected bugs and make code look closer to the way it runs. When using ES2015 or newer, one could use `const` and `let` which are block-scoped instead.

Reply:
Thanx, I wasn't aware of that!  Will certainly fix then.

Reply:
Yeah the `var enabled;` can be defined right above the top if(). That is how it is executed in js. 
This doesn't apply to `const` and `let`, but they have different scope too.